server/src/main/java/org/elasticsearch/common/lucene/search/XMoreLikeThis.java

```diff
@@ -727,6 +727,10 @@ private void addToQuery(PriorityQueue<ScoreTerm> q, BooleanQuery.Builder query)
         int numDocs = ir.numDocs();
         final int limit = Math.min(maxQueryTerms, words.size());
         FreqQ queue = new FreqQ(limit); // will order words by score
+        
+        if (limit == 0) {
+            return queue;
+        }
```

Comment:
Rather than checking for a particular value this late, I'd rather try improving the input validation for the `max_query_terms` parameter, both in the `MoreLikeThisQueryBuilder` setter and also maybe in MoreLikeThisQuery. I think we should reject negative or zero values here, since at least to my understanding it indicates a wrong usage of the API. Or are there cases you know where this parameter setting would make sense?

Reply:
I agree there is no sense to set ``max_query_terms`` = 0, maybe the API should validate this parameter instead.

Reply:
Are you interested in changing this PR to add validation to the `MoreLikeThisQueryBuilder` and `MoreLikeThisQuery` setters for the max query terms? I think we should reject all values <= 0 there already with an IllegalArgumentException and a fitting message. Adding these checkes should also be tests in the corresponding unit tests. If you don't want to do this just let me know, I'll open an issue so we can fix it later.

Reply:
Sure! thanks you